### ISSUE TITLE
APPEND result when targetSeq.Len() == 0

### ISSUE THREAD

**Issue by exander77** [ProtonMail/proton-bridge#2]
Already reported to support. Adding here as well.

Is this really ok, to return success if the sequence is empty? Seems to cover failure on the server.

```
  info := appendSucess
  if targetSeq.Len() > 0 {
    info = fmt.Sprintf("[%s %d %s] %s",
      appenduid,
      uidValidity,
      targetSeq.String(),
      appendSucess,
    )
  }

  return &imap.StatusResp{
    Type: imap.StatusOk,
    Info: info,
  }
```

My report:

> IMAP over ProtonBridge silently fails when importing mail into Sent folder where From corresponds to one of the user adresses causing a loss of emails.
> 
> Consider I have mail@radomirpolach.cz added to my account and minimal example below:
> 
> When I try to import a message like this where From corresponds to mail@mydomain.cz account:
> MIME-Version: 1.0
> Date: Mon, 10 Jan 2021 12:12:12 +0200
> Message-ID: <test@example.com>
> Subject: Test
> From: mail@mydomain.cz
> To: test@example.com
> 
> It results in this IMAP communication:
>   45:34.13 > b'KLII4 APPEND Sent "13-Apr-2020 01:45:34 +0200" {160}'
>   45:34.13 < b'+ send literal'
>   45:34.13 write literal size 160
>   45:34.13 < b'KLII4 OK APPEND completed'
> 
> Even though completed, the email is not added into the Sent folder.
> ProtonBridge does not report any error as well, just ordinary: No matching UID, continuing APPEND to Sent
> 
> When I on the other hand try to import a message where From corresponds to a different address not added to my account:
> MIME-Version: 1.0
> Date: Mon, 10 Jan 2021 12:12:12 +0200
> Message-ID: <test@example.com>
> Subject: Test
> From: mail2@mydomain.cz
> To: test@example.com
> 
> It results in this IMAP communication:
>   45:47.98 > b'OFJB4 APPEND Sent "13-Apr-2020 01:45:47 +0200" {161}'
>   45:47.98 < b'+ send literal'
>   45:47.98 write literal size 161
>   45:48.30 < b'OFJB4 OK [APPENDUID 4 33] APPEND successful'
> 
> And the email is stored in the Sent folder.
> ProtonBridge reports: Importing external message
> 
> This does not affect Inbox, only on Sent.

**Comment by exander77** [comment]
But this really needs an assessment from guys from ProtonMail.

**Comment by cuthix** [comment]
There were plenty reports that users see duplicates in `Sent` mailbox so we decided to filter `APPEND`s into `Sent` mailbox when sender is identical to account address and silently sends OK without import. We have plan to improve this in meanwhile you can try to append to different folder than `Sent`

For more details you can check the code:  
https://github.com/ProtonMail/proton-bridge/blob/master/internal/imap/mailbox_message.go#L113

**Comment by cuthix** [comment]
reference in GODT-143

**Comment by exander77** [comment]
@cuthix That's a scary fix. Anybody who tries to move mails into Sent **will lose them** and probably won't even notice. It may be a solution for me to move into another folder, but it is not a solution at all in a general sense.

Please deploy a fix as soon as possible and if I may advise you: never return success when the action didn't actually succeed as this could have catastrophic side effects.

**It destroyed almost 5 hundreds of my emails which I needed to recover from backup.**

**Comment by exander77** [comment]
> There were plenty reports that users see duplicates in Sent mailbox so we decided to filter APPENDs into Sent mailbox when sender is identical to account address and silently sends OK without import.

I am rereading this, I am really terrified.

**Comment by exander77** [comment]
I suggest fix for this to be deployed ASAP and for ProtonMail to inform all Bridge users not to use the affected versions. It is extremely easy to lose emails and don't even notice. I would even recommend advising users not to copy anything into Send folder until this is fixed.

**Comment by exander77** [comment]
I am starting to worry that nothing so far has been done with this issue which causes data loss.

**Comment by d310n9** [comment]
**TLDR: We bumped the priority on reevaluating this behavior and will try to find a better solution which prevents this scenario from happening at all in the coming sprints
More details:** 

Thanks again for flagging this issue, apologies for the pain you've encountered. 
As per previous discussion we acknowledge this behavior is frustrating and can lead to  message loss under the following conditions
a/ when trying to import/append to sent folder, _AND_
b/ when the sender of the imported emails = the email address of the 
(if either a/ or b/ does not apply, importing will work successfully) 

To give you some background on why this was done in the first place: 
1./ So far Bridge was primarily designed as an email processing client for sending & receiving mail, and not designed for importing. However now that Bridge has larger adoption, we do see the need for and want to provide users the ability to import reliably through it as well. 

2./ For send receive use case: Outlook on Windows (which is a significant portion of email client userbase) has an interesting behaviour where, after sending a message and trying to append the recently sent message to the send folder (as other email clients) **this appended message uses a different messageID than what was previously used during drafts and sending** (outlook specific) 

Long story short, if Bridge doesn't do anything, all Outlook users will see duplicate messages in the sent folder, which could erroneously be interpreted as "Bridge sent my message twice". Given that we prioritized send & receive over import workflow in the past, we built the feature as it works today. 
Going forward we'll be working on a solution which does both (i.e. functions as expected Outlook send but also allows users to successfully import to the sent folder for all cases). If you have some ideas, we'd be happy to discuss as well.

**Comment by exander77** [comment]
@d310n9 Regarding Outlook, isn't this an intended behaviour for using 3rd party SMTP server? New Outlook has an opt-in option "Save copies of messages in the Sent folder" or in older versions opt-out option "Do not save copies of sent items". Is this what you are talking about? If so, then you are actually breaking even intended and configurable Outlook behaviour. I think the different Message-ID is intended as well to prevent behaviour you are actually trying to do.

![save-sent-items-imap](https://user-images.githubusercontent.com/2256039/81059473-c654a800-8ed0-11ea-9e40-a2b040766d97.png)

![save-sent-options](https://user-images.githubusercontent.com/2256039/81059490-c9e82f00-8ed0-11ea-91f1-a336adb2443b.png)

I understand that the Bridge become more than it was, but I would point out that there lurks a bad practice. The software should not lie. It should not report success when it failed. It is ok if it fails for some corner cases, it is unpleasant, but it happens. But I have already found two instances when Proton Bridge outright lies:
1) This issue. And, it can result in data loss.
2) When an email is deleted from All Mail folder, it pops out again, because All Mail contains also Thash (which is extremely non-standard and breaking). This causes complete schisma between the mail client and Proton Bridge and me to have to remove cache from Proton Bridge and account from the mail client (ThunderBird).

**Comment by exander77** [comment]
ThunderBird has this feature using this plugin: https://addons.thunderbird.net/en-US/thunderbird/addon/copy-sent-to-current/

So you are probably breaking that as well.

**Comment by exander77** [comment]
It this fixed? I am afraid to test it.

**Comment by skooda** [comment]
Fixed by changing this behaviour in v1.2.8
